src/sentry_transport.c

```diff
@@ -10,11 +10,12 @@
 
 typedef struct sentry_transport_s {
     void (*send_envelope_func)(sentry_envelope_t *envelope, void *state);
-    void (*startup_func)(const sentry_options_t *options, void *state);
+    bool (*startup_func)(const sentry_options_t *options, void *state);
     bool (*shutdown_func)(uint64_t timeout, void *state);
     void (*free_func)(void *state);
     size_t (*dump_func)(sentry_run_t *run, void *state);
     void *state;
+    bool was_started;
```

Comment:
I'm curious if this should be `running`, and we set this to `false` after shutting down. The other option would be to empty out the transport so that everything afterwards becomes a noop.